src/vs/editor/contrib/suggest/suggestController.ts

```diff
@@ -427,6 +433,32 @@ export class SuggestController implements IEditorContribution {
 
 		// clear only now - after all tasks are done
 		Promise.all(tasks).finally(() => {
+
+			const resourceLoggedEvents = this._loggedEvents.get(model.uri) ?? new Set<string>();
+			if (!this._loggedEvents.has(model.uri)) {
+				this._loggedEvents.set(model.uri, resourceLoggedEvents);
```

Comment:
This is never being cleared so a potential, slow memory leak. It's also not clear why we gate the telemetry? You won't be able to say anything about the frequency of provider-usage because a single usage is enough, right?

Reply:
fyi - to reduce telemetry load (in case that's the intention here) I have used this pattern in the past: https://github.com/microsoft/vscode/blob/a704a69394a3fbd241feaa6c0305067f25f866f5/src/vs/editor/contrib/suggest/suggestModel.ts#L554. Basically only report telemetry every so many times

Reply:
Trying to reduce cycles that don't immediately benefit the user in the core edit loop as much as possible, the gating strategy makes good sense.